portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -64,18 +70,32 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         page = Integer.parseInt(pageString);
     }
     
-    int offset = page*numResults;
+    int offset = Math.max(0, page*numResults);
+
+    double deadline = 5.0;
+
+    // Construct a read policy for eventual consistency
+    ReadPolicy policy = new ReadPolicy(ReadPolicy.Consistency.EVENTUAL);
+
+    // Set the read policy
+    DatastoreServiceConfig eventuallyConsistentConfig =
+    DatastoreServiceConfig.Builder.withReadPolicy(policy);
+
+    // Set the call deadline
+    DatastoreServiceConfig deadlineConfig = DatastoreServiceConfig.Builder.withDeadline(deadline);
+
+    // Set both the read policy and the call deadline
+    DatastoreServiceConfig datastoreConfig =
+    DatastoreServiceConfig.Builder.withReadPolicy(policy).deadline(deadline);
 
     Query query = new Query("Messages").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
-    List<Entity> entities = results.asList(FetchOptions.Builder.withLimit(offset + numResults));
+    List<Entity> entities = results.asList(FetchOptions.Builder.withLimit(numResults).offset(offset));
 
-    if (entities.size() == 0 || 0 < offset+numResults && offset < entities.size())
-    {
-      comments = new ArrayList<Comment>();
-    }
-    for (int i = offset; i < offset+numResults && i < entities.size() && i >= 0; i++) {
+
+    comments = new ArrayList<Comment>();
+    for (int i = 0; i < numResults && i < entities.size() && i >= 0; i++) {
```

Comment:
The "i" starts from 0 and increments with "i++", it should always >= 0, right? So the "&& i>=0" is redundent.

Reply:
I did not notice that before, but thank you for pointing that out!